### ISSUE TITLE
Server Crashes When building project due to high memory usage

### ISSUE THREAD

**Issue by smit245** [ent/ent#3865]
I was readding [this](https://github.com/ent/ent/issues/3814) issue in closed issue section and i was facing the same issue in my project as well
We are unable to build the project in 16GB RAM.
It takes more then hour in local machine mostly.
On server it doesn't even compile and runs out of memory.

Suggestion
When we tried to build our project with -v argument then it takes too much RAM for  gen (gen = ent generated code folder) folder compilation or sometimes it takes too much RAM in some other folder which is using gen folder
so,
if we can move the schema table specific files from the gen directory to respective schema table folders in gen then it will not compile all together and memory usage will be reduced

In gen folder probably 5 files are generated per schema table and we have around 257 schema tables in database and we have generated 5 files per table so there will be more then 257*5=1285 files in gen folder only and there is a mutation.go file which takes so much time and RAM(my local machine hangs with SSD and 16GB RAM) just to open it in the editor(IntelliJ).

So if we can move the specific files to their respective folder then it might not compile all together and may compile folder by folder so it will take lesser RAM to compile

Go Version 1.21
Driver [MySql](https://github.com/go-sql-driver/mysql)
Ent 0.12.4

**Comment by nibbleshift** [comment]
Have you found a resolution to this?  I am having the same issue.

**Comment by mkraft** [comment]
We're experiencing the same problem which manifests when we `go test ./...`. It can only complete on machines with about >=32GB of RAM, otherwise freezing on a compile command that includes the Ent generated code.

**Comment by gebes** [comment]
Can be easily resolved when giving enough swap space

**Comment by tomleecode** [comment]
when using swap space, very slow

**Comment by gebes** [comment]
@tomleecode yeah, but it's currently the only solution. The ent framework is designed in such a way that you can't just refactor it easily. I tried myself, and everything has to be in one big folder to avoid circular dependencies. I think this is one of the biggest concerns about the future of ent, especially for enterprise customers with 300+ tables in a common scenario. This makes maintenance of go backends, which are supposed to compile swiftly, extremely difficult, even on beefy machines.

The problem is that many companies and developers will start using it, and eventually, they will be unable to migrate to anything else because they are already locked in and productivity will go down by a lot sadly

@a8m any comment?

**Comment by nibbleshift** [comment]
We abandoned ent just for this reason.  Not a feasible solution for
projects with a large number of data models.

With a current state of ent, you're probably better off with something like
sqlc.

**Comment by gebes** [comment]
@nibbleshift really? If you compare the development effort sqlc > entc imo so not worth it switching
